woodwork/indexers.py

```diff
@@ -45,3 +47,52 @@ def __getitem__(self, key):
         else:
             # singular value
             return selection
+
+
+class _iLocIndexerAccessor:
+    def __init__(self, data):
+        self.data = data
+        if dd and isinstance(self.data, dd.DataFrame):
+            raise TypeError("iloc is not supported for Dask DataFrames")
+        elif dd and isinstance(data, dd.Series):
+            raise TypeError("iloc is not supported for Dask Series")
+
+    def __getitem__(self, key):
+        selection = self.data.iloc[key]
+        return _process_selection(selection, self.data)
+
+
+class _locIndexerAccessor:
+    def __init__(self, data):
+        self.data = data
+
+    def __getitem__(self, key):
+        selection = self.data.loc[key]
+        return _process_selection(selection, self.data)
+
+
+def _process_selection(selection, original_data):
+    if isinstance(selection, pd.Series) or (ks and isinstance(selection, ks.Series)):
+        # col_name = selection.name
+        # if isinstance(self.ww_data, ww.DataTable) and set(selection.index.values) == set(self.ww_data.columns):
+        #     # return selection as series if series of one row.
+        #     return selection
+        # if isinstance(self.ww_data, ww.DataTable):
+        #     logical_type = self.ww_data.logical_types.get(col_name, None)
+        #     semantic_tags = self.ww_data.semantic_tags.get(col_name, None)
+        # else:
+        logical_type = original_data.ww.logical_type or None
+        semantic_tags = original_data.ww.semantic_tags.copy() or None
+        # if semantic_tags is not None:
+        #     semantic_tags = semantic_tags - {'index'} - {'time_index'}
+        selection.ww.init(logical_type=logical_type,
```

Comment:
Would it be possible to do what we do in `_make_series_call`?
```python
                        schema = copy.deepcopy(self._schema)
                        # We don't need to pass dtype from the schema to init
                        del schema['dtype']
                        result.ww.init(**schema)
```
I figure lines 84 and 85 may make it not possible here, but thought I'd suggest it

Reply:
I was actually thinking it is some of the uncommented code above that will complicate using this approach (when we are pulling a series off a dataframe), but I can take a closer look.

Reply:
Okay! If we do end up leaving it like this, we'll just need to remember to update it if the parameters to the Series Accessor init change (probably why we don't have description or metadata on the DataColumn tbh)

Maybe we could add a test like `test_get_subset_schema_all_params` that makes sure we test with all possible params to the init.

Reply:
@tamargrey I think your original suggestion here is the better approach. Changing for now and will worry about the commented code later when we get there.

Also, making this change highlighted an issue with the `_make_series_call` code when we have `use_standard_tags=False`, so I fixed that as well and added a new test to cover that situation.